plonky2x/core/src/backend/function/mod.rs

```diff
@@ -81,36 +89,32 @@ impl<C: Circuit> Plonky2xFunction for C {
             let contract_path = format!("{}/FunctionVerifier.sol", args.build_dir);
             let mut contract_file = File::create(&contract_path).unwrap();
 
-            let circuit_digest_bytes = circuit
+            info!("First building wrapper circuit to get the wrapper circuit digest...");
+            // The wrapper circuit digest will get saved in the Solidity smart contract,
+            // which will use this value as a public input `VerifierDigest` in the Gnark plonky2 verifier
+            let wrapped_circuit = WrappedCircuit::<L, WrapperParameters, D>::build(circuit);
+
+            let mut circuit_digest_bytes = wrapped_circuit
```

Comment:
We should standardize the circuit_digest calculation because it's used in a lot of other places e.g., knowing the name to use when saving things to disk.

Maybe it can be a method on top of `CircuitBuild`.

Reply:
Including the .reverse() part

Reply:
In this case, the circuit_digest calculation is actually a bit different because the element is a `Hash` which lives in a different field depending on the hasher...